### ISSUE TITLE
MLP not working properly when dealing with a multiclass classification problem

### ISSUE THREAD

**Issue by rafaelgreca** [rafaelgreca/scratchml#13]
I observed that this problem occurs when trying to train an MLP model on a non-binary dataset. The error is due to the use of the Softmax activation function, which is causing the gradient to vanish (in earlier epochs the loss tends to be infinite, and then the weights turns to nan values).
